### PR TITLE
design refresh: change dark mode selected blob lines background color

### LINKED ISSUE #22185: Dark mode file highlights lack contrast
## Problem: dark mode file highlights lack contrast

image

## Solution: $grey-12 becomes the new dark mode highlight color

image

## Figma:

https://www.figma.com/file/eCdwOD2Xsot8M91oZt0Ukj/Dark-mode-file-highlights?node-id=8%3A0&viewport=434%2C396%2C0.4098958373069763

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #22185

### CODE REVIEW

**Review by pdubroy — APPROVED** [review]

**Comment by felixfbecker** [comment]
It worries me that the `--color-bg-4` is not a suitable color variable that adapts to light/dark theme and we need a dark-theme-specific override, which I thought we want to move away from. @rrhyne would it be possible to instead update the dark variant of `--color-bg-4`? Or perhaps pick a different `bg` color that fits and adapts automatically?

**Comment by rrhyne** [comment]
@felixfbecker @novoselrok we don't have a --color-bg-4 in the design system. I'm thinking that value is left over from the old system. I believe FPT is just now removing the old colors, which I did not expect to be the case, so I'm a bit worried about how many of the old colors are used throughout the application. @umpox can you weigh in here?

As for the correct color here, we are not using bg colors in this application.  In the design system, we are using $grey-03 to specify this value for light mode and $gray-12 in dark. If you would rather have a specific semantic color, that's possible, but it has some issues. 

This single use doesn't warrant a background color, IMO... plus we would have to shuffle names around as this color is lighter than the current-03. I don't think we should do this.

We have top options I can see:

* Create a semantic color for this (--code-selection?). 
  * I'm not a fan as it would only be used once.
* Use gray values
  * I believe FTP is ok with this approach)

**Comment by felixfbecker** [comment]
Thanks for the response Rob!
I'd personally rather see variables introduced even if they are just used in one place than referencing static colors because it will make custom theming harder down the line, but I'll leave it to FPT.

**Comment by rrhyne** [comment]
A code selection semantic color could be a good call to support different syntax themes. I could see comments being unreadable in some themes without changing that value.

**Comment by novoselrok** [comment]
@rrhyne @felixfbecker Hopefully, I understood your comments correctly, so I added a semantic color `--code-selection-bg`, with the following definition:
```scss
.theme-light.theme-redesign {
   // ...
    --code-selection-bg: var(--gray-03);
}

.theme-dark.theme-redesign {
   // ...
    --code-selection-bg: var(--gray-12);
}

```

**Comment by rrhyne** [comment]
@novoselrok  can we get a screenshot in this PR?

**Comment by novoselrok** [comment]
> @novoselrok can we get a screenshot in this PR?

@rrhyne Sure, here they are:

Screenshot 2021-06-23 at 17 54 56
Screenshot 2021-06-23 at 17 55 13

**Review by limitedmage — APPROVED** [review]
Code LGTM as long as design is OK with color change.

**Comment by rvantonder** [comment]
Looking at that screenshot, and after clarifying in [thread](https://sourcegraph.slack.com/archives/C0W2E592M/p1624382795292900?thread_ts=1623874449.250500&cid=C0W2E592M) we should be using a lighter color to highlight in dark mode, not a darker one. I personally think using a darker color for highlighting in dark mode has the effective of de-emphasizing, rather than emphasizing the selection. Please comment/request changes on this PR if that is accurate @rrhyne.

File: client/branded/src/global-styles/colors-redesign.scss [hunk 1]
```diff
@@ -175,6 +176,7 @@ $theme-colors-redesign: (
     --code-line-highlight-color: var(--body-bg);
     --diff-add-bg: #224035;
     --diff-remove-bg: #3e1d1d;
+    --code-selection-bg: var(--gray-12);
```
Comment (rrhyne) [inline client/branded/src/global-styles/colors-redesign.scss:179]:
```suggestion
    --code-selection-bg: var(--gray-09);
```

Comment (rrhyne) [reply client/branded/src/global-styles/colors-redesign.scss:179]: @novoselrok this comment addresses @rvantonder's comment and the slack thread. I erroneously stated grey-12 was the color to use, when in fact it was gray-09. 

Apologies!!!!

Comment (rrhyne) [reply client/branded/src/global-styles/colors-redesign.scss:179]:
```suggestion
    --code-selection-bg: var(--gray-08);
```

**Comment by rrhyne** [comment]
Apologies all... I've provided the wrong value to Rok and made a few alterations after seeing it in product. The final value will be gray-08. Here's the screenshot:

image

**Review by rrhyne — APPROVED** [review]
Final looks great with gray-08 instead of my previous recommendations.

**Pull request merged**